File: sentinel/agents/core/tests/fixtures/multilang/Fibonacci.java
public class Fibonacci {
    public static int fibonacci(int n) {
        if (n <= 0) return 0;
        if (n == 1) return 1;
        int a = 0, b = 1;
        for (int i = 2; i <= n; i++) {
            int temp = b;
            b = a + b;
            a = temp;
        }
        return b;
    }

    public static String greet(String name) {
        if (name == null || name.isEmpty()) {
            return "Hello, World!";
        }
        return "Hello, " + name + "!";
    }
}

File: sentinel/agents/core/tests/fixtures/multilang/fibonacci.rs
fn fibonacci(n: u64) -> u64 {
    if n <= 0 {
        return 0;
    }
    if n == 1 {
        return 1;
    }
    let (mut a, mut b) = (0u64, 1u64);
    for _ in 2..=n {
        let temp = b;
        b = a + b;
        a = temp;
    }
    b
}

fn greet(name: &str) -> String {
    if name.is_empty() {
        return String::from("Hello, World!");
    }
    format!("Hello, {}!", name)
}
